File: 01-programming/csharp/concepts/code/01-basics/syntax-comments.cs
// -o-o-o-o- Syntax (Comments) -o-o-o-o-

/* This file goes over the different types of comments in C# simply.
 *
 *  # # # # KEY # # # #
 *
 * Comments are lines of text ignored by the compiler. They are used to explain
 * code, leave notes, or temporarily disable code. C# supports the same comment
 * styles as C, plus one extra type for documentation.
 *
 * // - single-line comment. Everything after // on that line is ignored.
 *      Same as in C.
 *
 * /* ... * / - multi-line comment. Everything between the opening and closing
 *      markers is ignored, across as many lines as needed.
 *      Same as in C.
 *
 * /// - XML documentation comment. Unique to C#. Used to describe classes,
 *       methods, and parameters. Tools like Visual Studio and IDEs read these
 *       to generate tooltips and documentation automatically.
 *       No equivalent in standard C.
 *
 * Note: Comments do not affect how the program runs.
 *       Good comments explain WHY, not just WHAT.
 *
 * */

using System;

class Program {

    /// <summary>
    /// This is an XML documentation comment.
    /// It describes what the method does for IDEs and documentation tools.
    /// </summary>
    /// <param name="name">The name to greet.</param>
    /// <returns>A greeting string.</returns>
    static string Greet(string name) {
        return $"Hello, {name}!"; // inline comment: returns a greeting
    }

    static void Main() {

        // This is a single-line comment
        Console.WriteLine("Comments do not affect output.");

        /*
         * This is a multi-line comment.
         * It can span as many lines as needed.
         * Useful for longer explanations or temporarily disabling code.
         */
        Console.WriteLine("Multi-line comments work the same as in C.");

        // Calling the documented method
        string message = Greet("World");
        Console.WriteLine(message);

        // Console.WriteLine("This line is commented out and will not run.");

    }
}

File: 01-programming/python/concepts/code/01-basics/syntax-comments.py
# -o-o-o-o- Syntax (Comments) -o-o-o-o-

# This file goes over the different types of comments in Python simply.
#
#  # # # # KEY # # # #
#
# #         - single-line comment. Everything after # on that line is ignored.
#             Equivalent to // in C and C#.
#             There is no /* */ multi-line comment symbol in Python.
#
# """ ... """ or ''' ... ''' - a multi-line string used as a comment.
#             Technically these are string literals that are not assigned
#             to anything, so Python ignores them. They are widely used
#             as multi-line comments and as docstrings (see below).
#             Equivalent to /* */ in C and C#.
#
# docstring  - a """ string placed immediately inside a function, class,
#              or module to describe what it does. Python can read these
#              at runtime using the __doc__ attribute.
#              Similar to /// XML doc comments in C#.
#              No equivalent in C.
#
# Note: Good comments explain WHY, not just WHAT.
#       Comments do not affect how the program runs.
#

# This is a single-line comment
print("Comments do not affect output.")

"""
This is a multi-line string used as a comment.
It can span as many lines as needed.
Python ignores it because it is not assigned to anything.
"""

print("Multi-line string comments work fine in Python.")


def greet(name):
    """
    This is a docstring — it describes the function.
    It is readable at runtime via greet.__doc__
    
    Parameters:
        name (str): The name to greet.
    
    Returns:
        str: A greeting string.
    """
    return f"Hello, {name}!"  # inline comment: returns a greeting


message = greet("World")
print(message)

# You can read a docstring at runtime
print(greet.__doc__)

# print("This line is commented out and will not run.")
